internal/core/src/segcore/SkipIndex.h

```diff
@@ -0,0 +1,246 @@
+// Copyright (C) 2019-2020 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License
+
+
+#pragma once
+#include <unordered_map>
+
+#include "common/Types.h"
+#include "log/Log.h"
+
+namespace milvus{
+union MinValue {
+    int8_t int8Value;
+    int16_t int16Value;
+    int32_t int32Value;
+    int64_t int64Value;
+    float floatValue;
+    double doubleValue;
+};
+
+union MaxValue {
+    int8_t int8Value;
+    int16_t int16Value;
+    int32_t int32Value;
+    int64_t int64Value;
+    float floatValue;
+    double doubleValue;
+};
+
+struct FieldChunkMetrics {
+    MinValue min;
+    MaxValue max;
+    bool hasValue_;
+
+    FieldChunkMetrics() : hasValue_(false){};
+};
+
```

Comment:
SkipIndex is a separate class and can be extended for using fieldChunkMetrics in various ways.